Code/Tools/ProjectManager/Source/ProjectManagerSettings.cpp

```diff
@@ -14,46 +14,235 @@
 
-    QString GetProjectBuiltSuccessfullyKey(const QString& projectName)
-    {
-        return QString("%1/Projects/%2/BuiltSuccessfully").arg(ProjectManagerKeyPrefix).arg(projectName);
-    }
+                if (saveToDisk)
+                {
+                    SaveProjectManagerSettings();
+                }
+            }
 
-    QString GetExternalLinkWarningKey()
-    {
-        return QString("%1/SkipExternalLinkWarning").arg(ProjectManagerKeyPrefix);
-    }
-}
+            return success;
+        }
+
+        bool RemoveProjectManagerKey(const QString& settingsKey, bool saveToDisk)
+        {
+            bool success = false;
+            auto settingsRegistry = AZ::SettingsRegistry::Get();
+            if (settingsRegistry)
+            {
+                success = settingsRegistry->Remove(settingsKey.toStdString().c_str());
+
+                if (saveToDisk)
+                {
+                    SaveProjectManagerSettings();
+                }
+            }
+
+            return success;
+        }
+
+        bool CopyProjectManagerKeyString(const QString& settingsKeyOrig, const QString& settingsKeyDest, bool removeOrig, bool saveToDisk)
+        {
+            bool success = false;
+            auto settingsRegistry = AZ::SettingsRegistry::Get();
+            if (settingsRegistry)
+            {
+                AZStd::string settingsValue;
+                success = settingsRegistry->Get(settingsValue, settingsKeyOrig.toStdString().c_str());
+
+                if (success)
+                {
+                    success = settingsRegistry->Set(settingsKeyDest.toStdString().c_str(), settingsValue);
+                    if (success)
+                    {
+                        if (removeOrig)
+                        {
+                            success = settingsRegistry->Remove(settingsKeyOrig.toStdString().c_str());
+                        }
+
+                        if (saveToDisk)
+                        {
+                            SaveProjectManagerSettings();
+                        }
+                    }
+                }
+            }
+
+            return success;
+        }
+
+        QString GetProjectKey(const ProjectInfo& projectInfo)
+        {
+            return QString("%1/Projects/%2/%3").arg(ProjectManagerKeyPrefix, projectInfo.m_id, projectInfo.m_projectName);
+        }
+
+        QString GetExternalLinkWarningKey()
+        {
+            return QString("%1/SkipExternalLinkWarning").arg(ProjectManagerKeyPrefix);
+        }
+
+        QString GetProjectsBuiltSuccessfullyKey()
+        {
+            return QString("%1/SuccessfulBuildPaths").arg(ProjectManagerKeyPrefix);
+        }
+
+        bool GetBuiltSuccessfullyPaths(AZStd::set<AZStd::string>& result)
+        {
+            bool success = false;
+            auto settingsRegistry = AZ::SettingsRegistry::Get();
+            if (settingsRegistry)
+            {
+                QString builtKey = GetProjectsBuiltSuccessfullyKey();
```

Comment:
If you used the static constexpr char above then you wouldn't need to create this QString